packages/components/src/base-control/index.js

```diff
@@ -8,36 +8,47 @@ import classnames from 'classnames';
  */
 import VisuallyHidden from '../visually-hidden';
 
-function BaseControl( { id, label, hideLabelFromVision, help, className, children } ) {
+function BaseControl( { as = 'div', id, label, hideLabelFromVision, help, className, children } ) {
+	const isFieldSet = as === 'fieldset';
```

Comment:
I personally wonder if we're trying to force two distinct components into one. 

 - A Fieldset is in general a wrapper to "controls" and it has a legend
 - A Control is a form element + label rendered inside (or not) a fieldset.

Should we just use separate components instead?